base/threadingconstructs.jl

```diff
@@ -110,15 +111,73 @@ A barrier is placed at the end of the loop which waits for all tasks to finish
 execution.
 
 The `schedule` argument can be used to request a particular scheduling policy.
-The only currently supported value is `:static`, which creates one task per thread
-and divides the iterations equally among them. Specifying `:static` is an error
-if used from inside another `@threads` loop or from a thread other than 1.
+
+Except for `:static` scheduling, how the iterations are assigned to tasks, and how the tasks
+are assigned to the worker threads are undefined. The exact assignments can be different
+for each execution. The scheduling option should be considered as a hint. The loop body
+code (including any code transitively called from it) must not assume the task and worker
+thread in which they are executed. The loop body code for each iteration must be able to
+make forward progress and be free from data races independent of the state of other iterations.
+As such, synchronizations across iterations may invoke deadlock.
+
+For example, the above conditions imply that:
+
+- The lock taken in an iteration must be released within the same iteration.
+- Avoid communication between iterations using, e.g., `Channel`s.
+- Write only to locations not shared across iterations (unless a lock or atomic operation is used).
+
+Furthermore, even though `lock` and atomic operations can be useful sometimes, it is often better
+to avoid them for performance.
+
+Schedule options are:
+- `:static` creates one task per thread and divides the iterations equally among
+            them, assigning each task specifically to each thread.
+            Specifying `:static` is an error if used from inside another `@threads` loop
+            or from a thread other than 1.
+- `:dynamic` tries to schedule iterations dynamically to available worker threads,
+            assuming that the workload for each iteration is uniform.
+
+If no schedule is specified, when called from thread 1 the default is `:static`, or when
```

Comment:
```suggestion
If no schedule is specified the behavior is dependent on the calling thread, when called from thread 1 the default is `:static`, when
```

Reply:
```suggestion
Without the scheduler argument, the exact scheduling is unspecified; i.e., it may be
different across Julia releases. Currently, the behavior is dependent on the calling thread.
When called from thread 1 the default is `:static`. When
```